### PR TITLE
chore: create universal binaries for OS X (10.15 and higher)

### PR DESCRIPTION
**Pull request by rjaegers**
Build a universal binaries supporting both x86_64 and arm64 architectures.
Set the minimum supported target version to 10.15.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,11 @@
 cmake_minimum_required(VERSION 3.24)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
+if (CMAKE_SYSTME_NAME STREQUAL "Darwin")
```
Comment (richardapeters) [inline CMakeLists.txt:4]:
```suggestion
if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+    set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "OS X target architectures")
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14" CACHE STRING "Minimum OS X deployment version")
```
Comment (richardapeters) [inline CMakeLists.txt:6]: This version says 10.14 but the feature title says 10.15. Which one is it?

**Review by richardapeters — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (rjaegers) [reply CMakeLists.txt]: 🤡

File: CMakeLists.txt (hunk 2 above)
Comment (rjaegers) [reply CMakeLists.txt]: 🤡

**Pull request merged**